src/components/AuthorViewpoint/FactListCard.tsx

```diff
@@ -0,0 +1,117 @@
+"use client";
+import { Fact } from "@/types/conversations.types";
+import EditViewpointFact from "@/components/AuthorViewpoint/EditViewpointFact";
+import { MagnifyingGlassIcon, PlusIcon } from "@heroicons/react/24/outline";
+import { Select, Button } from "@mantine/core";
+import { useState, useEffect, Dispatch, SetStateAction } from "react";
+import { allFacts } from "@/mock/conversationMock";
+
+type FactListCardProps = {
+    viewpointFactList: Fact[];
+    setViewpointFactList: Dispatch<SetStateAction<Fact[]>>;
+};
+
+export default function FactListCard({
+    viewpointFactList,
+    setViewpointFactList,
+}: FactListCardProps) {
+    const [searchData, setSearchData] = useState<Fact[]>(allFacts);
+    const [selectedFactId, setSelectedFactId] = useState<string>("");
+    const [searchValue, setSearchValue] = useState<string>(""); // eslint-disable-line
+
+    useEffect(() => {
+        const selectedFact = searchData.find(
+            (fact) => String(fact.id) === selectedFactId,
+        );
+        if (selectedFact) {
+            //move the selected fact to the viewpointFactList
+            setViewpointFactList((prev) => [...prev, selectedFact]);
+
+            //remove the selected fact from the searchData
+            const newSearchData = searchData.filter(
```

Comment:
This approach is not quite practical when actualing integrating the backend, as filtering out the search result array will fail once a new network resquest overrides the original data.